src/MailHide.php

```diff
@@ -280,7 +280,7 @@ public function setEmail($email)
         }
 
         $this->emailLocalPart  = $emailParts[0];
-        $this->emailDomainPart = $emailParts[1];
+        $this->emailDomainPart = $emailParts[1] ?? '';
```

Comment:
This change seems wrong: the second key must always exist.

If this was done for static analysis, I suggest reverting and adding to baseline instead.

Reply:
Hi @Ocramius,

As you per your suggestion, we're planning to change the statement. So, the fix (as given in [Psalm](https://psalm.dev/docs/running_psalm/issues/PossiblyUndefinedArrayOffset/) ) could possibly be like:

From:
```
        $this->emailDomainPart = $emailParts[1] ?? '';
```
To:
```
        if(!isset($emailParts[1])) {
            $emailParts[1] = "";
        }

        $this->emailLocalPart  = $emailParts[0];
        $this->emailDomainPart = $emailParts[1];
```

For the below error of  QA Checks (Psalm:undefined array keys) :
```
PossiblyUndefinedArrayOffset
src/MailHide.php:283:34: PossiblyUndefinedArrayOffset: Possibly undefined array key $emailParts[1]
```

Could you please review and provide your inputs on the same ?

Reply:
@glo17680 I suggested expanding the baseline with no code changes to the original source.

Reply:
We have made the suggested changes.